cmake/BipedalLocomotionFrameworkDependencies.cmake

```diff
@@ -124,3 +133,19 @@ framework_dependent_option(FRAMEWORK_COMPILE_MasImuTest
 framework_dependent_option(FRAMEWORK_COMPILE_JointTrajectoryPlayer
   "Compile joint-trajectory-player application?" ON
   "FRAMEWORK_COMPILE_YarpImplementation;FRAMEWORK_COMPILE_Planners;FRAMEWORK_COMPILE_RobotInterface;FRAMEWORK_COMPILE_matioCppConversions;FRAMEWORK_USE_matioCpp;FRAMEWORK_USE_YARP" OFF)
+  
+framework_dependent_option(FRAMEWORK_COMPILE_Perception
+  "Compile Perception libraries?" ON
+  "FRAMEWORK_USE_OpenCV;FRAMEWORK_USE_PCL" OFF)
+  
+framework_dependent_option(FRAMEWORK_COMPILE_PerceptionInterface
+  "Compile PerceptionInterface libraries?" ON
+  "FRAMEWORK_COMPILE_Perception" OFF)
+
+framework_dependent_option(FRAMEWORK_COMPILE_RealsenseCapture
+  "Compile Realsense related software?" ON
+  "FRAMEWORK_COMPILE_PerceptionInterface;FRAMEWORK_USE_realsense2" OFF)
+  
+framework_dependent_option(FRAMEWORK_COMPILE_RealSenseTestApplication
+  "Compile realsense-test application?" ON
+  "FRAMEWORK_COMPILE_Perception;FRAMEWORK_COMPILE_RealsenseCapture;FRAMEWORK_COMPILE_PerceptionInterface" OFF)
```

Comment:
missing a FRAMEWORK_COMPILE_YarpImplementation

Reply:
Wonder how the CMake step in CI is not failing for macOS and Windows ? @traversaro would you have any advice on this?

Reply:
I think the steps that fails is "Check build if some dependencies are not enabled " https://github.com/dic-iit/bipedal-locomotion-framework/blob/master/.github/workflows/ci.yml#L401 , that runs just on Ubuntu as this kind of bugs are platform indipendent?

Reply:
Ok, I think I only partially understand. But, since it's platform independent, shouldn't an error have risen even without trying to disable some dependencies in the case where `FRAMEWORK_COMPILE_YarpImplementation` was not available and the target links to some YARP related libraries. 

I was trying to follow the cmake macros, but I'm afraid I am missing to make some logical explanation to myself related to `framework_dependent_option`.